### PR TITLE
feat: API command

### PR DESCRIPTION
**Pull request by sustained**
# Todo

- [x] Compile data from API docs into useable structured data.
- [x] Add `APIService`.
- [x] Add `!api` lookup command.
- [x] Add aliases.
- [x] Add fuzzy searching.

# Future?

- [ ] Add vuex, vue-router.
- [ ] Vue 2 and Vue 3 API support.

# See

https://vuejs.org/v2/api/

### CODE REVIEW

**Comment by sustained** [comment]
This is ready, in my opinion.

**Note:** It's actually only ~200 lines of code, the other 1100 are a `HJSON` file containing the data extracted from the [API docs](https://vuejs.org/v2/api/).

# Notes

- I used HJSON because the file was handwritten.
- It doesn't account for typos yet (however it supports aliases) but there are changes from other PRs that can help us with that (relating to [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance) and such).
- In future we can look into extracting this data from the Markdown by parsing the Abstract Syntax Tree but I attempted this and gave up due to several challenges that I don't want to deal with right now.
- In future we want the same kind of thing for vuex and vue-router imo.

**Comment by sustained** [comment]
It now accounts for typos (fuzzy search added).

This is complete unless we want to add vuex and vue-router too already but I think we should see how this feature goes (if people like it, we can expand it later).

Marking as ready to review.

File: src/commands/documentation/api.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+      }
+
+      if (!api || api.length === 0) {
+        throw new APINotFoundError(
+          `Could not find anything in the API matching "${lookup}".`
+        )
+      }
+
+      const reply = await msg.channel.send(EMPTY_MESSAGE, { embed })
+      tryDelete(msg, 7500)
```
Comment (lloydjatkinson) [inline src/commands/documentation/api.js]: How did you settle on these values? We should really make these config settings, I think.

Comment (sustained) [reply src/commands/documentation/api.js]: The values were essentially made up by me on the spot.

I agree but I'd like to tackle this in another PR.

The reason for that being that all the PRs have code like this - it doesn't make sense to implement that in this PR, agreed?

See #38.

**Comment by sustained** [comment]
@Elfayer So @lloydjatkinson has reviewed this one and said it was OK, think we can merge?

**Pull request merged**